### PR TITLE
CRS-3421 Refactoring and remove deprecations 

### PR DESCRIPTION
**Pull request by raghavtan**
Refactoring code
Remove deprecated output mechanism from action
Updated tests  
Move dependabot out of workflows

### CODE REVIEW

**Comment by raghavtan** [comment]
fixes #22

**Review by gmsantos — APPROVED** [review]

**Comment by raghavtan** [comment]
Will create a v3 release and tag with this

File: neo/neo.py [hunk 1]
```diff
@@ -9,37 +9,38 @@
 import json
 import re
 from urllib.parse import quote_plus
-from typing import List
+from typing import List, Any
+import subprocess
```
Comment (klemmster) [inline neo/neo.py:13]:
```suggestion
```
I think subprocess isn't used anywhere.

File: neo/neo.py [hunk 2]
```diff
@@ -61,49 +63,62 @@ def update_matches(files: List[str]):
...
-    defaults: List[str] = [],
-    default_patterns: List[str] = [],
-    per_page: int = 0,
-):
+        github_token: str,
+        github_repository: str,
+        github_base_ref: str,
+        github_head_ref: str,
+        include_regex: str,
+        defaults=None,
```
Comment (klemmster) [inline neo/neo.py:85]: defaults is a bool, at least according to the action.yml input
`defaults:bool = False`

File: neo/neo.py [hunk 3]
```diff
@@ -61,49 +63,62 @@ def update_matches(files: List[str]):
...
+    :param github_base_ref:
+    :param github_head_ref:
+    :param include_regex:
+    :param defaults:
+    :param default_patterns:
+    :param per_page:
+    :return:
+    """
+    default_patterns = [] if default_patterns is None else default_patterns
+    defaults = [] if defaults is None else defaults
```
Comment (klemmster) [inline neo/neo.py:90]: remove this line, defaults is a boolean.
if you applied above suggestion, it defaults to false :)

File: neo/neo.py [hunk 4]
```diff
@@ -21,7 +21,7 @@ def update_matches(files, include_regex):
         if match:
             if match.groupdict():
                 if "reason" in match.groupdict().keys():
-                    raise ValueError("reason is a reserved name for the job matrix")
+                    raise ValueError(eval("reason is a reserved name for the job matrix"))
```
Comment (klemmster) [inline neo/neo.py:25]: Why do  we eval the string?

**Review by klemmster — COMMENTED** [review]
Thank you for splitting up the tests and the other useful changes.
My pyright spottet some type related issues, that would be great if you could address them.

File: neo/neo.py (hunk 1 above)
Comment (raghavtan) [reply neo/neo.py:13]: Already fixed

File: neo/neo.py (hunk 2 above)
Comment (raghavtan) [reply neo/neo.py:85]: this was already fixed

File: neo/neo.py [hunk 5]
```diff
@@ -9,37 +9,38 @@
...
-    include_regex: str,
-    defaults=False,
-    default_patterns=[],
-    default_dir=os.getenv("GITHUB_WORKSPACE", os.curdir),
-):
+        files: list[dict[str, str]],
+        include_regex: str,
+        defaults: bool = False,
+        default_patterns: object = None,
+        default_dir: object = os.getenv("GITHUB_WORKSPACE", os.curdir)) -> list[Any]:
```
[the remark this answers is not in the document]
Comment (raghavtan) [reply neo/neo.py:38]: fixed #26

File: neo/neo.py (hunk 4 above)
Comment (raghavtan) [reply neo/neo.py:25]: fixed #26

File: neo/neo.py (hunk 3 above)
Comment (raghavtan) [reply neo/neo.py:90]: fixed #26

File: neo/neo.py (hunk 2 above)
Comment (raghavtan) [reply neo/neo.py:85]: fixed #26

**Pull request merged**